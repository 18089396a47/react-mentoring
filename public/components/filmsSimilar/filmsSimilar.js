import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const FilmsSimilar = ({ title }) => (
    <span>{title ? `Similar as ${title}` : null}</span>
);

FilmsSimilar.propTypes = {
    title: PropTypes.string
};

const mapStateToProps = (state) => {
    return {
        title: state.filmInfo.title
    };
};

export default connect(mapStateToProps)(FilmsSimilar);
