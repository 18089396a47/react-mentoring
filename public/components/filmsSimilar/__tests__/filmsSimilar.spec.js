import React from 'react';
import renderer from 'react-test-renderer';
import { FilmsSimilar } from '../filmsSimilar';


describe('FilmsSimilar component', () => {
    it('FilmsSimilar should show empty state', () => {
        const tree = renderer.create(
            <FilmsSimilar title="" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('FilmsSimilar should show film title', () => {
        const tree = renderer.create(
            <FilmsSimilar title="test title" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
