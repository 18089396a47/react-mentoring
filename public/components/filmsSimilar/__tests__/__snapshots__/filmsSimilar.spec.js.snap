// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsSimilar component FilmsSimilar should show empty state 1`] = `<span />`;

exports[`FilmsSimilar component FilmsSimilar should show film title 1`] = `
<span>
  Similar as test title
</span>
`;
